pkg/codegen/pcl/binder_component.go

```diff
@@ -135,10 +135,31 @@ func ComponentProgramBinderFromFileSystem() ComponentProgramBinder {
 			DirPath(componentSourceDir),
 			ComponentBinder(ComponentProgramBinderFromFileSystem()))
 
+		includeSourceDirectoryInDiagnostics(programDiags, componentSourceDir)
+
 		return componentProgram, programDiags, err
 	}
 }
 
+// Replaces the Subject of the input diagnostics from just a file name to the full path of the component directory.
+func includeSourceDirectoryInDiagnostics(diags hcl.Diagnostics, componentSourceDir string) {
+	for _, diag := range diags {
+		start := hcl.Pos{}
+		end := hcl.Pos{}
+
+		if diag.Subject != nil {
+			start = diag.Subject.Start
+			end = diag.Subject.End
+		}
+
+		diag.Subject = &hcl.Range{
+			Filename: componentSourceDir,
```

Comment:
Shouldn't this include the original filename as well?

Reply:
Yeah you are right, I added that in latest commit ✅